### PR TITLE
Add source metadata for `vceregen`

### LINKED ISSUE #3886: Add source metadata for vceregen data
This is an easy task to do before we get any access to the raw data. Just fill out the `pudl/metadata/sources.py` for this table. This metadata is used by the `pudl-archiver` to understand the data is it trying to archive.

It is used in the programmatic construction of the `datapackage.json` which accompanies our archived data on Zenodo, and which is necessary for the PUDL datastore script to retrieve and verify a particular partition of the dataset.

### PR DESCRIPTION
**Pull request by aesharpe**
# Overview

Closes #3886

## What did you change?

Adds source metadata for `vceregen` so it's ready to be ingested by the archiver.

### CODE REVIEW

File: src/pudl/metadata/sources.py
Comment (aesharpe) [inline src/pudl/metadata/sources.py:846]: Based on the description of `gridpathtoolkit`

File: src/pudl/metadata/sources.py
Comment (aesharpe) [inline src/pudl/metadata/sources.py:840]: Was not sure which link to put here because the data are proprietary. Could have linked to either of these PDFs as well: 

https://vibrantcleanenergy.com/wp-content/uploads/2021/04/2018%20VCE%20Study_Dataset%20Methods%20and%20Analysis536960.pdf 

https://vibrantcleanenergy.com/wp-content/uploads/2020/08/VCE-Weather-Dataset-Overview_August2020.pdf

File: src/pudl/metadata/sources.py
Comment (aesharpe) [inline src/pudl/metadata/sources.py:867]: same as `gridpathtoolkit` but with `vibrant clean energy` and `county` added

File: src/pudl/metadata/sources.py
Comment (aesharpe) [inline src/pudl/metadata/sources.py:870]: I assumed it would be the same as `gridpathtoolkit`

**Review by e-belfer — APPROVED** [review]
I have a question for @zaneselvans about the licensing.

We might want to add the documentation PDFs somewhere - left some thoughts in the comments.

File: src/pudl/metadata/sources.py
Comment (zaneselvans) [reply src/pudl/metadata/sources.py:840]: I asked Chris in email which was the appropriate document and he said the [Dataset Overview](https://vibrantcleanenergy.com/wp-content/uploads/2020/08/VCE-Weather-Dataset-Overview_August2020.pdf) (Aug, 2020) was the better one to reference.  I also asked permission to archive it alongside the inputs (or the way we do in our GitHub repo) so that when the VCE website goes down it's still available.

File: src/pudl/metadata/sources.py
Comment (zaneselvans) [reply src/pudl/metadata/sources.py:870]: Yes, we got them to agree to CC-BY-4.0 after a bit of discussion. We need to make sure that the dataset is clearly attributed to VCE in the Zenodo archive. And also in the metadata that accompanies the PUDL outputs. This isn't something we've had to deal with in the past because all the input data has come from the US Govt which doesn't have copyrights.

File: src/pudl/metadata/sources.py
Comment (e-belfer) [reply src/pudl/metadata/sources.py:840]: Will we be able to commit a large PDF to git, space-wise?

File: src/pudl/metadata/sources.py
Comment (e-belfer) [reply src/pudl/metadata/sources.py:870]: Ah, that's not in the licensing, that's in the contributors - we need to define a new organization in the CONTRIBUTOR field for VCE and define it here.

Comment (zaneselvans) [reply src/pudl/metadata/sources.py:870]: Yeah, that sounds right!  I think so long as VCE has the "creator" role or whatever is equivalent, there's a CC-BY-4.0 license, they're mentioned explicitly in the description (which ends up in the description on Zenodo) and linked to as well, and in the Data Sources page in our docs, and we get analogous appropriate metadata into our Data Release archives on Zenodo, we'll be okay.

**Review by e-belfer — CHANGES_REQUESTED** [review]
We should add VCE in the CONTRIBUTORS dictionary and make them the contributor of this dataset, as discussed in the comments. Also worth seeing if we can check in the key PDF into our data sources folder in Git.

File: src/pudl/metadata/sources.py
Comment (aesharpe) [reply src/pudl/metadata/sources.py:840]: Has he given us the go-ahead to archive the PDF or are we still waiting for a response?

Comment (e-belfer) [reply src/pudl/metadata/sources.py:840]: It's public on the website so I don't see why it would be a problem.

File: src/pudl/metadata/sources.py
Comment (aesharpe) [inline src/pudl/metadata/sources.py:841]: Could make this a path to the internal file, but wasn't 100% sure how to do that.

**Review by e-belfer — APPROVED** [review]
Looks good!

File: src/pudl/metadata/constants.py
Comment (zaneselvans) [inline src/pudl/metadata/constants.py:196]: Once we've got an archiver that works for a private file handoff, it'd be good to re-archive the GridPath RA Toolkit data using it, and do something similar for attribution / licensing there.

Comment (aesharpe) [reply src/pudl/metadata/constants.py:196]: @e-belfer

Comment (e-belfer) [reply src/pudl/metadata/constants.py:196]: Definitely, though slightly tricky to refresh the metadata without new files but it can be done. Want to write up an issue so it doesn't get lost?

Comment (zaneselvans) [reply src/pudl/metadata/constants.py:196]: See #3895

**Pull request merged**
